contracts/earn/strategies/Platypus/StrategyPlatypusVector.sol

```diff
@@ -0,0 +1,224 @@
+pragma solidity 0.6.12;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import '@openzeppelin/contracts/math/SafeMath.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/utils/Pausable.sol';
+import '@uniswap/lib/contracts/libraries/TransferHelper.sol';
+
+import '../../../interfaces/VectorFinance/IPoolHelper.sol';
+import '../../../interfaces/VectorFinance/IMainStaking.sol';
+import '../../../interfaces/BankerJoe/IJoeRouter.sol';
+import '../../../interfaces/IWooAccessManager.sol';
+import '../../../interfaces/IWETH.sol';
+import '../BaseStrategy.sol';
+
+contract StrategyPlatypusVector is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ----- State Variables ----- */
+
+    // DepositPool list:
+    // usdc.e pool helper: 0x257D69AA678e0A8DA6DFDA6A16CdF2052A460b45
+    //
+    IPoolHelper public poolHelper;
+    address public mainStaking = address(0x8B3d9F0017FA369cD8C164D0Cc078bf4cA588aE5);
+
+    address[] public reward1ToWantRoute;
+    address[] public reward2ToWantRoute;
+    uint256 public lastHarvest;
+    uint256 public slippage = 100; // 100 = 1%; 300 = 3%; 1 = 0.01%
+
+    /* ----- Constant Variables ----- */
+
+    address public constant wrappedEther = address(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7); // WAVAX
+
+    address public constant reward1 = address(0xe6E7e03b60c0F8DaAE5Db98B03831610A60FfE1B); // VTX
+    address public constant reward2 = address(0x22d4002028f537599bE9f666d1c4Fa138522f9c8); // PTP
+    address public constant uniRouter = address(0x60aE616a2155Ee3d9A68541Ba4544862310933d4); // JoeRouter02
+
+    /* ----- Events ----- */
+
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+
+    constructor(
+        address initVault,
+        address initAccessManager,
+        address initPoolHelper,
+        address[] memory initReward1ToWantRoute,
+        address[] memory initReward2ToWantRoute
+    ) public BaseStrategy(initVault, initAccessManager) {
+        poolHelper = IPoolHelper(initPoolHelper);
+        reward1ToWantRoute = initReward1ToWantRoute;
+        reward2ToWantRoute = initReward2ToWantRoute;
+
+        require(
+            reward1ToWantRoute.length > 0 && reward1ToWantRoute[reward1ToWantRoute.length - 1] == want,
+            'StrategyPlatypusVector: !route'
+        );
+        require(
+            reward2ToWantRoute.length > 0 && reward2ToWantRoute[reward2ToWantRoute.length - 1] == want,
+            'StrategyPlatypusVector: !route'
+        );
+
+        _giveAllowances();
+    }
+
+    function e_withdraw(IERC20 _token) external onlyAdmin returns (uint256 amount) {
+        amount = _token.balanceOf(address(this));
+        if (amount > 0) {
+            _token.safeTransfer(owner(), amount);
+        }
+    }
```

Comment:
Maybe add `require(address(_token) != want);` first?
Need to move this func to `/* ----- Admin Functions ----- */`

Reply:
This is only for debug. I'm removing it.